### PR TITLE
make extra_hosts values uniques

### PR DESCRIPTION
**Pull request by glours**
Issue occurred when you have an `extra_hosts` value defined in the main config file and you have an `extends` of the service with the exact same name without any additional `extra_hosts` config

Fixes https://github.com/docker/compose/issues/12155

### CODE REVIEW

File: override/uncity.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func init() {
 	unique["services.*.environment"] = keyValueIndexer
 	unique["services.*.env_file"] = envFileIndexer
 	unique["services.*.expose"] = exposeIndexer
+	unique["services.*.extra_hosts"] = keyValueIndexer
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply override/uncity.go:53]: one option could be to drop strictly equal lines, which somehow relax the json schema unicity rule, which doesn't bring any value here

[the remark this answers is not in the document]
Comment (glours) [reply override/uncity.go:53]: Ok, I'll build a custom indexer

File: override/merge.go [hunk 2]
```diff
@@ -163,6 +163,22 @@ func mergeNetworks(c any, o any, path tree.Path) (any, error) {
...
+	// Rewrite content of left slice to remove duplicate elements
+	i := 0
+	for _, v := range left {
+		if !slices.Contains(right, v) {
+			left[i] = v
+			i++
+		}
+	}
+	// keep only not duplicated elements from left slice
+	left = left[:i]
```
Comment (ndeloof) [inline override/merge.go:178]: you can use https://pkg.go.dev/slices#Compact to remove duplicates from a slice

**Review by ndeloof — APPROVED** [review]

**Comment by Nek-** [comment]
Thanks for the bugfix! Is there an estimated date for a new release? 😅

**Comment by jhrotko** [comment]
@Nek- We are not sure, but should be around in 1~2 weeks

**Pull request merged**